test/PayID/fakes/fake-pay-id-client.ts

```diff
@@ -0,0 +1,21 @@
+import PayIDClientInterface from '../../../src/PayID/pay-id-client-interface'
+import Result from '../../Common/Helpers/result'
+
+/**
+ * A fake Pay ID Client which can return faked values.
+ */
+export default class FakePayIDClient implements PayIDClientInterface {
+  /**
+   * @param xrpAddressResult The object that will be returned or thrown from a call to `xrpAddressForPayID`.
+   */
+  constructor(private readonly xrpAddressResult: result<string | undefined>) {}
+
+  // eslint-disable-next-line @typescript-eslint/require-await
+  async xrpAddressForPayID(_payID: string): Promise<string | undefined> {
```

Comment:
Is there a case where `xrpAddressResult` is `undefined` as opposed to being an instance of `Error`?

Wondering if the return type should be an "either monad" on `string | undefined`, or just `string`

Reply:
#283 changes this API.